src/main/kotlin/io/meshcloud/dockerosb/persistence/GitHandlerService.kt

```diff
@@ -121,7 +121,13 @@ open class GitHandlerService(
   private fun fetchMergePush(): Boolean {
     // fetch from the remote
     git.fetch()
-        .apply { remote = "origin" }
+        .apply {              
+          remote = "origin" 
+
+          gitConfig.username?.let {
```

Comment:
discuss: for my taste we started copy/pasting this around a a little too much by now and i think it makes sense to move this into a common method called by all git operations (so we don't miss HTTPS basic auth support on other git operations by accident)